### PR TITLE
Coverage reports

### PR DESCRIPTION
**Pull request by 1AmNegan**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change
Add test coverage reports for unit, integration and feature tests.

### CODE REVIEW

File: .coveralls.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+repo-token: 52OLLNCnPFH2EgHdDBFz19cOGsLAmmhri
```
Comment (prasadtalasila) [inline .coveralls.yml]: coveralls does not require repo token on travis. You can add this file to .gitignore.

File: package.json [hunk 2]
```diff
@@ -4,10 +4,13 @@
   "description": "Codebase for backward incompatible revision of autolabcli v0.1.1.",
   "main": "index.js",
   "scripts": {
-    "posttest": "nyc report --reporter=lcov > coverage.lcov && codecov",
+    "posttest": "nyc report --reporter=lcov > coverage.lcov",
```
Comment (prasadtalasila) [inline package.json]: We might not require this line.

File: .coveralls.yml (hunk 1 above)
Comment (prasadtalasila) [reply .coveralls.yml]: We can't have the repo token on public repository for security reasons. Any one with that token can upload code coverage to our codecov project.

File: .travis.yml [hunk 3]
```diff
@@ -7,5 +7,6 @@ install:
 
 script:
   - npm run unit
-  - npm run posttest
+  - npm run post-unit
   - npm run integration
+  - npm run post-integration
```
Comment (prasadtalasila) [inline .travis.yml]: The last line can be 
`  - npm run post-integration || :`

We need the extra bit to make sure that this command does not fail on travis.

Comment (1AmNegan) [reply .travis.yml]: This failed only in the travis build during PR. It worked fine in my commit.

Comment (1AmNegan) [reply .travis.yml]: Do I need to include `:` too?

Comment (prasadtalasila) [reply .travis.yml]: Yes. The ":" is essential. It is equivalent to "no operation" which always succeeds.

**Pull request merged**